apis/projectcontour/v1/detailedconditions.go

```diff
@@ -113,4 +113,20 @@ type DetailedCondition struct {
 	Warnings []SubCondition `json:"warnings,omitempty"`
 }
 
-const ValidConditionType string = "Valid"
+const (
+	ValidConditionType string = "Valid"
+
+	IncludeError         = "IncludeError"
```

Comment:
I can't make these a type since that breaks the API being the helpers aren't in `internal` packages so this was left as strings. 

Should we name these with `ConditionType` appended? For example, s/IncludeError/IncludeErrorConditionType/ ?

Reply:
I might do `ConditionTypeIncludeError` - helps group them, particularly for code completion.

Reply:
Alternately we could add a method per condition type - e.g. `.AddIncludeError(...)`, `.AddAuthError(...)`, etc. Perhaps makes it less likely that a developer will inadvertently pass an invalid condition type?

Reply:
I think maybe let's just do the strings for now, could revisit the helpers. The thing about the helpers is they get out of hand quickly, like how the `HTTPConnectionManagerBuilder` is today inside `envoy/listener.go`